src/hotspot/share/opto/library_call.cpp

```diff
@@ -5915,8 +5915,13 @@ bool LibraryCallKit::inline_arraycopy() {
     record_for_igvn(slow_region);
 
     // (1) src and dest are arrays.
+    // Keep track of the information that src/dest are arrays to prevent below array specific accesses from floating above.
     generate_non_array_guard(load_object_klass(src), slow_region);
+    const Type* tary = TypeAryPtr::make(TypePtr::BotPTR, TypeAry::make(Type::BOTTOM, TypeInt::POS), nullptr, false, Type::OffsetBot);
```

Comment:
Is this never used else where? Should it a static field in `TypeAryPtr` same as `TypeAryPtr::BYTES` and friends?

Reply:
I wondered as well and no, we don't use this type anywhere else (the closest would be `TypeAryPtr::RANGE`). We only create it when meeting arrays of primitive and non-primitive element type. Do you think this should still go to `TypeAryPtr::*`?

Reply:
I would add it to `TypeAryPtr` (maybe as `TypeAryPtr::BOTTOM`) . The main benefit I see is that the new code would more readable if it referred to `TypeAryPtr::BOTTOM` rather than the long type creation expression.

Reply:
Sounds good, I'll update the patch accordingly.